### PR TITLE
Add an FAQ to the developer guide for new contributors

### PR DESCRIPTION
**Pull request by rossbar**
This is an attempt to address some common questions to help new contributors get started contributing to networkx.

### CODE REVIEW

File: doc/developer/faq.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+.. code-block:: bash
+
+   git checkout -b complete-graph-circular-layout-example
+
+.. note:: It's generally a good idea to give your branch a descriptive name so
+   that it's easy to remember what you are working on.
+
+Now you can begin work on your example. Sticking with the circular layout idea,
+you might create a file in ``examples/drawing`` called ``circular_layout.py``
```
Comment (jarrodmillman) [inline doc/developer/faq.rst:52]: Should probably be called ``plot_circular_layout.py``.

Comment (rossbar) [reply doc/developer/faq.rst:52]: Good catch!

**Comment by rossbar** [comment]
Rebased to fix pygeos doc building issue

File: doc/developer/faq.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+   In [2]: nx.kamada_kawai_layout?
+
+.. code-block:: text
+
+   Signature: <clipped for brevity>
+   Docstring: <clipped for brevity>
+   File: ~/networkx/networkx/drawing/layout.py
+   Type: function
+
+The ``grep`` command-line utility is also very useful. For example, from the
```
Comment (stefanv) [inline doc/developer/faq.rst:104]: Not sure if `git grep` deserves a mention here; but may be useful for Windows developers.

**Review by stefanv — COMMENTED** [review]
Nice, thanks Ross!

File: doc/developer/faq.rst [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  1. The source code
+  2. A development environment
+  3. An idea of what you'd like to contribute
+
+Steps 1 & 2 are covered extensively in :ref:`Development Workflow <dev_workflow>`.
+There is no generic answer for step 3. There are many ways that NetworkX can
+be improved, from adding new algorithms, improving existing algorithms,
+improving the test suite (e.g. increasing test coverage), and improving the
+documentation.
+The "best" way to find a place to start is to follow your own personal
```
[the remark this answers is not in the document]
Comment (rossbar) [reply doc/developer/faq.rst:23]: There are a couple things that we could point to: the "Good first issue" label is an obvious one; there's also the "Sprint" label which indicates issues that, in principle, are small enough to be tackled within a several-hour timeframe; finally, there's the [Algorithms category](https://github.com/networkx/networkx/discussions/categories/algorithms) in GitHub discussions, which includes a few posts on algorithms that might be nice to add to NX.

I'll add some bullets on these to try to improve the guidance here - other ideas of things to highlight are more than welcome!

**Review by stefanv — APPROVED** [review]

**Comment by jarrodmillman** [comment]
Maybe, we should rename  ``doc/developer/faq.rst`` to ``doc/developer/getting_started.rst``; but keep the FAQ format.  I think we should keep this document focused on novice contributors to open source (and I worry that just calling this FAQ will lead to a mix of topics some for novice contributors and some for experienced contributors).

So, for example, a shorted discussion about adding a gallery example should appear in ``CONTRIBUTING.rst``.  But the longer discussion about adding gallery examples could stay in ``doc/developer/getting_started.rst``.

That way we can keep ``CONTRIBUTING.rst`` short and focused on folks familiar with contributing to open source scientific Python project (but not necessarily to NetworkX) who just want the minimal commands to run.   And the ``doc/developer/getting_started.rst`` document can focus on people new to  contributing to open source scientific Python project in general.

I suspect folks new to NetworkX (but not new to open source) will find a very short contributor guide more helpful.  They are also likely to fork, clone the repo, and open ``CONTRIBUTING.rst`` from their terminal.  For folks new to open source, are more likely to appear on the mailing list or on issues.  So we can point them to ``doc/developer/getting_started.rst``, which is a combination FAQ and roadmap for folks new to open source.

Maybe we should also include pointers (I imaging a sentence or just a bullet point) to things like:
- learning Git (e.g., https://git-scm.com/doc, https://try.github.io/)
- learning GitHub (e.g., https://docs.github.com/en/github/getting-started-with-github/set-up-git)

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -3,6 +3,13 @@
 Contributor Guide
 =================
 
+.. note::
+   This document assumes some familiarity with contributing to open source
+   scientific Python projects using GitHub pull requests. If this does not
+   describe you, you may first want to see the :ref:`contributing_faq`.
```
Comment (dschult) [inline CONTRIBUTING.rst:9]: Should this reference be to `new_contributor_faq` now instead of `contributing_faq`?

Comment (rossbar) [reply CONTRIBUTING.rst:9]: In this case the `contributing_faq` is an explicitly specified link target (see the `.. _contributing_faq:` on the first line of `new_contributor_faq.rst` doc), so this link should work (and continue to work) regardless of the name of the file or the title of the document.

**Review by dschult — APPROVED** [review]

**Pull request merged**